GCR/composite.py

```diff
@@ -17,10 +17,15 @@
 MATCHING_FORMAT = None
 MATCHING_ORDER = tuple()
 
-
-def _match(index_this, index_main, sorter=None):
-    if sorter is None:
+def _match(index_this, index_main, sorter=None, selector=None, select_descending=False):
+    if (sorter is None) and (selector is None):
         sorter = np.argsort(index_this)
+    elif sorter is None:
+        sorter = np.argsort(selector)
+        if select_descending:
+            sorter = sorter[::-1]
+        sorter2 = np.argsort(index_this[sorter], kind='stable')
+        sorter = sorter[sorter2]
```

Comment:
suggest this nested if so that it's more clear when sorter will be calculated
```python
    if sorter is None:
        if selector is None:
            sorter = np.argsort(index_this)
        else:
            sorter = np.argsort(selector)
            if select_descending:
                sorter = sorter[::-1]
            sorter2 = np.argsort(index_this[selector_sorter], kind='stable')
            sorter = sorter[sorter2]
```

Reply:
good idea! that looks much cleaner